core/src/test/java/com/datastax/oss/driver/internal/core/config/map/MapBasedDriverConfigLoaderTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.core.config.map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.datastax.oss.driver.api.core.config.DefaultDriverOption;
+import com.datastax.oss.driver.api.core.config.DriverConfig;
+import com.datastax.oss.driver.api.core.config.DriverConfigLoader;
+import com.datastax.oss.driver.api.core.config.DriverExecutionProfile;
+import com.datastax.oss.driver.api.core.config.DriverOption;
+import com.datastax.oss.driver.internal.core.config.MockOptions;
+import com.datastax.oss.driver.internal.core.config.typesafe.DefaultDriverConfigLoader;
+import java.util.Map;
+import java.util.SortedSet;
+import java.util.concurrent.ConcurrentHashMap;
+import org.junit.Test;
+
+public class MapBasedDriverConfigLoaderTest {
+
+  @Test
+  public void should_reflect_changes_in_real_time() {
+    Map<String, Map<DriverOption, Object>> optionsMap = new ConcurrentHashMap<>();
+    Map<DriverOption, Object> defaultProfileMap = new ConcurrentHashMap<>();
+    optionsMap.put(DriverExecutionProfile.DEFAULT_NAME, defaultProfileMap);
+    defaultProfileMap.put(MockOptions.INT1, 1);
+
+    DriverConfigLoader loader = DriverConfigLoader.fromMap(optionsMap);
+    DriverConfig config = loader.getInitialConfig();
+    assertThat(config.getDefaultProfile().getInt(MockOptions.INT1)).isEqualTo(1);
+
+    defaultProfileMap.put(MockOptions.INT1, 2);
+    assertThat(config.getDefaultProfile().getInt(MockOptions.INT1)).isEqualTo(2);
+  }
+
+  /**
+   * Checks that, if we ask to pre-fill the default profile, then we get the same set of options as
+   * the built-in reference.conf.
+   */
+  @Test
+  public void should_fill_default_profile_like_reference_file() {
```

Comment:
This test will protect us against `buildDefaultOptionsMap()` getting out of sync with `reference.conf`.

Reply:
This is a very good idea. :+1: